src/main/java/com/umc/networkingService/domain/schedule/service/ScheduleServiceImpl.java

```diff
@@ -0,0 +1,101 @@
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ScheduleServiceImpl implements ScheduleService {
+    private final ScheduleRepository scheduleRepository;
+    private final ScheduleMapper scheduleMapper;
+
+    @Override
+    public ScheduleInfoSummariesInCalendar getCalendarByMonth(Long month) {
+
+
+        return scheduleMapper.toScheduleInfoSummariesInCalendar(
+                scheduleRepository.findSchedulesByMonth(month).stream()
+                .map(schedule -> scheduleMapper.toScheduleInfoSummaryInCalendar(schedule))
+                .toList());
+    }
+
+    @Override
+    public ScheduleInfoSummaryLists getScheduleLists(Long month) {
+        List<Schedule> schedulesLists = scheduleRepository.findSchedulesByMonth(month);
+        List<ScheduleInfoSummary> campusSchedules = schedulesLists.stream()
+                .filter(schedule -> schedule.getHostType().equals(HostType.CAMPUS))
+                .map(schedule -> scheduleMapper.toScheduleInfoSummary(schedule))
+                .toList();
+
+        List<ScheduleInfoSummary> branchSchedules = schedulesLists.stream()
+                .filter(schedule -> schedule.getHostType().equals(HostType.BRANCH))
+                .map(schedule -> scheduleMapper.toScheduleInfoSummary(schedule))
+                .toList();
+
+        List<ScheduleInfoSummary> centerSchedules = schedulesLists.stream()
+                .map(schedule -> scheduleMapper.toScheduleInfoSummary(schedule))
+                .filter(schedule -> schedule.getHostType().equals(HostType.CENTER))
+                .toList();
+
+        return scheduleMapper.toScheduleInfoSummaries(campusSchedules, branchSchedules, centerSchedules);
+    }
+
+    @Override
+    public ScheduleDetail getScheduleDetail(Member member, UUID scheduleId) {
+        Schedule schedule = scheduleRepository.findById(scheduleId).orElseThrow(() -> new RestApiException(ErrorCode.EMPTY_SCHEDULE));
+
+        return scheduleMapper.toScheduleDetail(schedule);
+    }
+
+    @Override
+    public ScheduleId createSchedule(Member member, CreateSchedule request) {
+        Schedule schedule = scheduleMapper.createScheduleToSchedule(member, request);
+
+        scheduleRepository.save(schedule);
+
+        return scheduleMapper.UUIDtoScheduleId(schedule.getId());
+    }
+
+    @Override
+    public ScheduleId updateSchedule(Member member, UUID scheduleId, UpdateSchedule request) {
+        Schedule schedule = scheduleRepository.findById(scheduleId).orElseThrow(() -> new RestApiException(ErrorCode.EMPTY_SCHEDULE));
+        // 만약 수정하려는 멤버와 일정 작성자가 일치하지 않을 경우 에러 반환
+        if (!schedule.getWriter().getId().equals(member.getId())) {
+            throw new RestApiException(ErrorCode.NO_PERMISSION_MEMBER);
+        }
+        schedule = scheduleMapper.updateScheduleToSchedule(request, schedule);
+
+        scheduleRepository.save(schedule);
+
+        return scheduleMapper.UUIDtoScheduleId(schedule.getId());
+    }
```

Comment:
@Transactional 어노테이션을 추가하면 schedule 엔티티의 변경사항이 트랜잭션이 종료될 때 자동으로 DB에 반영되기 때문에 save 메소드를 호출하지 않아도 됩니다
```suggestion
    @Override
    @Transactional
    public ScheduleId updateSchedule(Member member, UUID scheduleId, UpdateSchedule request) {
        Schedule schedule = scheduleRepository.findById(scheduleId).orElseThrow(() -> new RestApiException(ErrorCode.EMPTY_SCHEDULE));
        // 만약 수정하려는 멤버와 일정 작성자가 일치하지 않을 경우 에러 반환
        if (!schedule.getWriter().getId().equals(member.getId())) {
            throw new RestApiException(ErrorCode.NO_PERMISSION_MEMBER);
        }
        schedule = scheduleMapper.updateScheduleToSchedule(request, schedule);
        
        return scheduleMapper.UUIDtoScheduleId(schedule.getId());
    }
```

Reply:
이거 그럼 createSchedule도 동일한가요???

Reply:
createSchedule의 경우는 새로운 엔티티 객체를 생성해서 저장을 직접 해주어야해요!!

Reply:
수정했습니다!